src-python/lock-keeper/lockkeeper.py

```diff
@@ -131,6 +132,32 @@ def fl_restart():
     return jsonify({'status': 'ok'})
 
 
+@app.route('/floodlight/tc', methods=['POST'])
+def floodlight_shape_traffic():
+    body = request.get_json()
+    common_commands = ['tc qdisc del dev eth0 root',
+                       'tc qdisc add dev eth0 root handle 1: prio priomap 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2',
+                       'tc qdisc add dev eth0 parent 1:1 handle 10: netem delay {}ms'.format(
+                           body['tcData']['egressDelayMs'])]
+    switches_by_container = defaultdict(list)
+    for sw in body['affectedAddresses']:
+        switches_by_container[sw['containerName']].append(sw)
+    for container, switches in switches_by_container.items():
+        commands = common_commands.copy()
+        for switch in switches:
+            commands.extend(['tc filter add dev eth0 protocol ip parent 1:0 prio 1 u32 match ip dst {}/32 match ip '
+                             'dport {} 0xffff flowid 1:1'.format(switch['ip'], switch['port'])])
+        execute_commands_in_container(commands, container)
+    return jsonify({'status': 'ok'})
```

Comment:
Do you mind extending text response(for virtual and hardware implementations)?
for instance: 
* after executing `execute_commands_in_container`
* create list([[containerNames, ip, port],] 
* return it in response
`"tc filter is applied in the following containers: {}".format(<generated_list>)`

I guess it can be useful during investigating.

Reply:
done